TF_Service_dotNet/TouchFree/Interactions/PositioningModule.cs

```diff
@@ -1,32 +1,32 @@
 ﻿using System;
-using System.Collections;
-using System.Collections.Generic;
-using Leap.Unity;
-using UnityEngine;
-using Ultraleap.TouchFree.ServiceShared;
+using System.Numerics;
+using System.Linq;
 
-namespace Ultraleap.TouchFree.Service
+using Leap;
+
+
+namespace Ultraleap.TouchFree.Library.Interactions
 {
-    public class PositioningModule : MonoBehaviour
+    public class PositioningModule
     {
         public TrackedPosition trackedPosition = TrackedPosition.INDEX_STABLE;
 
-        [Tooltip("If assigned, the cursor snapper and stabiliser will be accessed from the utils object.")]
-        public GameObject positioningUtils;
-
-        public PositionStabiliser Stabiliser;
-
         private Positions positions;
 
         private const float NEAREST_BONE_BIAS = 0.01f;
 
-        protected void OnEnable()
+        public readonly PositionStabiliser Stabiliser;
+
+        public PositioningModule(PositionStabiliser _stabiliser, TrackedPosition _position)
         {
-            if (positioningUtils != null)
-            {
-                Stabiliser = positioningUtils.GetComponent<PositionStabiliser>();
-            }
+            Stabiliser = _stabiliser;
+            trackedPosition = _position;
 
+            Enable();
+        }
+
+        protected void Enable()
+        {
             Stabiliser.ResetValues();
         }
```

Comment:
This could be in one line rather than having an Enable method?

also do we even need to reset the values if each interaction has its own PositionStabiliser?

Reply:
It's pulled out in the enable method because it will need to be called on each instance of switching between interactions, not _just_ on initialisation (which I admittedly haven't done any set-up of within TouchPlane (Enable/Disable behaviour) yet).

Also I _think_ the Stabiliser needs to be reset even with seperate Stabilizers because we don't want to try to stabilise new data based on old? Not sure it'll have a _huge_ effect as the stabilizer only caches one previous pos but it could cause some funky drift from the last position this interaction saw a hand to the actual hand pos if we don't reset.